### PR TITLE
[Feature] Supports viewing which applications a variable is referenced by

### LINKED ISSUE #1876: [Feature] Supports viewing which applications a variable is referenced by
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/streampark/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

Supports viewing which applications a variable is referenced by.
When a variable is about to be deleted, if the variable is being referenced by the application, it will prompt that it cannot be deleted, and the user needs to be able to see which are referenced by which application

### Usage Scenario

_No response_

### Related issues

[Proposal][Feature] Add variable management module #1477 
[Feature] Reference variables as placeholders in program args and Flink SQL #1781 

[you can check the thread](https://lists.apache.org/thread/s9sjpkjn6spvq07bk34cfd3vln7rjo0m)

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by muchunjin**
## What changes were proposed in this pull request

Issue Number: close #1876  

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

## Does this pull request potentially affect one of the following parts
 - Dependencies (does it add or upgrade a dependency): (yes / no)

### CODE REVIEW

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/mapper/ApplicationMapper.java [hunk 1]
```diff
@@ -38,7 +38,15 @@ public interface ApplicationMapper extends BaseMapper<Application> {
...
+    @Select("select a.*, " +
+            "       u.username, " +
+            "       case when trim(u.nick_name) = ''" +
+            "           then u.username" +
+            "           else u.nick_name" +
+            "           end as nick_name " +
+            "from t_flink_app a, t_user u " +
+            "where a.user_id = u.user_id " +
+            "   and a.team_id=#{teamId}")
     List<Application> getByTeamId(@Param("teamId") Long teamId);
```
Comment (wolfboys) [inline streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/mapper/ApplicationMapper.java:41]: I recommend write the query sql to mapper.xml

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/VariableServiceImpl.java [hunk 2]
```diff
@@ -138,26 +157,38 @@ public String replaceVariable(Long teamId, String mixed) {
     }
 
     private boolean isDependByApplications(Variable variable) {
-        // Detect whether the variable is dependent on the args of the application
+        if (CollectionUtils.isEmpty(getDependApplicationsByCode(variable))) {
+            return false;
+        }
+        return true;
```
Comment (wolfboys) [inline streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/VariableServiceImpl.java]: Code could be simpler: 
```
return CollectionUtils.isNotEmpty(getDependApplicationsByCode(variable));
```

File: streampark-console/streampark-console-webapp/src/views/system/variable/DependApps.vue [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+-->
+
+<template>
+  <a-card :bordered="false" style="margin-top: 20px;">
+    <a-descriptions
+      bordered
+      size="middle"
+      layout="vertical">
+      <template slot="title">
+        <span class="app-bar">List of applications dependent on the variable "{{ this.variable }}"</span>
```
Comment (wolfboys) [inline streampark-console/streampark-console-webapp/src/views/system/variable/DependApps.vue]: would it be better?
```
Variable "{{ this.variable }}" used list
```

**Comment by muchunjin** [comment]
@wolfboys Thanks for the hard review, I made changes according to your suggestion.

**Comment by wolfboys** [comment]
> @wolfboys Thanks for the hard review, I made changes according to your suggestion.

LGTM.

**Comment by wolfboys** [comment]
cc @wangsizhu0504 add a new page, please know

**Review by wolfboys — APPROVED** [review]
LGTM

**Pull request merged**